xml/bp_troubleshooting_objectgateway.xml

```diff
@@ -160,124 +158,46 @@ osdmap e9 pg 2.d2041a48 (2.0) -> up [1,0] acting [1,0]
-</screen>
-     <para>
-       This stops new metadata from being written to the
-       <literal>default.rgw.meta</literal> pool, but does not remove any existing objects or pool.
-     </para>
-   </sect2>
-   <sect2 xml:id="cleaning-metadata-pool-heap">
-     <title>Cleaning the Metadata Heap Pool</title>
-     <para>
-       Clusters created prior to <emphasis>Jewel</emphasis> normally use
-       the <literal>default.rgw.meta</literal> only for the metadata archival feature.
-     </para>
-     <para>
-       However, from <emphasis>Luminous</emphasis> onwards, radosgw uses pool
-       namespaces within <literal>default.rgw.meta</literal> for an entirely
-       different purpose. For example, to store <literal>user_keys</literal> and
-       other critical metadata.
-     </para>
-     <para>
-       Users should check zone configuration before proceeding any cleanup procedures:
-     </para>
+ <sect1 xml:id="large-omap-issues">
+  <title>Large OMAP Issues</title>
+  <para>
+    If you receive a cluster warning for a large OMAP issue, it means that
+    some object has exceeded one of the osd deep scrub large OMAP object *
+    threshold, usually set to 200K keys and 1G of storage. The relevant pool or PG
+    involved can be found by grepping for large OMAP object as mentioned in
+    the warning message. Depending on the pool and the object key involved
+    it might indicate one of the following:
+  </para>
+  <sect2 xml:id="resharding-issues">
+    <title>Resharding Issues</title>
+    <para>
+      If the large omap object name happens to be in the
+      <literal><replaceable>zone</replaceable>.rgw.buckets.index</literal> pool
+      this means that a bucket has more than 200K keys, dynamic resharding is
+      set on by default in single site clusters which would automatically have
+      resharded the bucket, for multi site clusters however this is not supported
+      and would have to be manually done <xref linkend="ogw-bucket-reshard"/>. Looking at the
+      bucket stats would reveal the total object count and the number of shards.
+      This is helpful to understand if the bucket has already been resharded,
+      in which case the warning is just because the deep scrub happened to run
+      before the reshard process and would be cleared in the next deep scrub.
+    </para>
+  </sect2>
+  <sect2 xml:id="usage-statistics">
+    <title>Usage Statistics</title>
+    <para>
+      Usage stats are also stored as omap keys in the <literal><replaceable>zone</replaceable>.rgw.log</literal>
+      pool, however these are not automatically trimmed, so a manual trimming
+      of usage stats would have to be done. For example:
+    </para>
 <screen>
-&prompt.cephuser.ogw;radosgw-admin zone get --rgw-zone=default | grep default.rgw.meta
-[should not match any strings]
+&prompt.smaster;radosgw-admin usage trim [--uid=<replaceable>user-id</replaceable>] --start-date=2019-01-01 --end-date=2019-03-03
```

Comment:
'smaster' prompt? or is it going to be run on salt master, or better on the admin node?

Reply:
Hmm probably admin node, sorry. I'll update.